internal/pkg/instrumentation/bpf/go.opentelemetry.io/otel/traceglobal/bpf/probe.bpf.c

```diff
@@ -388,6 +391,33 @@ static __always_inline long fill_tracer_id(tracer_id_t *tracer_id, go_tracer_ptr
     return 0;
 }
 
+// This instrumentation attaches uprobe to the following function:
+// func (t *tracer) newSpan(ctx context.Context, autoSpan *bool, name string, opts []trace.SpanStartOption) (context.Context, trace.Span) {
+// https://github.com/open-telemetry/opentelemetry-go/blob/ac386f383cdfc14f546b4e55e8726a0a45e8a409/internal/global/trace.go#L161
+SEC("uprobe/newSpan")
+int uprobe_newStart(struct pt_regs *ctx) {
```

Comment:
Ideally this would be unloaded once this sets the flag for a process.

It was suggested by @RonFed that we could send a special event to the output perf buffer and in the probe read the event as a special case and unload the probe. Investigation is needed.

Reply:
I've updated locally to accomplish this but having issues. The `parent` span in the e2e test fails to get tracing IDs. The debug tracing logs contain:

```
             app-2959921 [006] ...11 527565.795722: bpf_trace_printk: write_span_context: failed to write trace ID: -14
             app-2959921 [006] ...11 527565.795730: bpf_trace_printk: failed to write span context: -2
```

Looking at the kernel errors `-14` indicates this is likely `EFAULT`: https://elixir.bootlin.com/linux/v6.12.5/source/mm/maccess.c#L143

I'm not 100% sure that is where the error is coming from, and I'm very much less sure why the error is being trown.

Reply:
I'm considering just skipping this unload step and tracking in an issue for follow-up.

Reply:
@MrAlias could you push a commit with the attempt to unload the probe and I can have a look at it?

Reply:
@RonFed sure thing: https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/1405/commits/9a4811713b8e3385dd0fa0633c2178aa6622b716

Reply:
Hmm, things seem to be "working" now. I think the [upgrade of github.com/cilium/ebpf to v0.17.0](https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/1500) may have changed things.

Still need to look into the span kind failure.

Reply:
It does looks to be working locally as well. I'm not exactly sure what changed to make this work, but happy to see it is working as expected.

Reply:
Resolving. This looks resolved and ready for review.